src/main/java/core/basesyntax/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.security;
+
+import core.basesyntax.exception.AuthenticationException;
+import core.basesyntax.lib.Inject;
+import core.basesyntax.lib.Service;
+import core.basesyntax.model.Driver;
+import core.basesyntax.service.DriverService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Driver driver = driverService.findByLogin(login)
+                .orElseThrow(() -> new AuthenticationException("Incorrect login. "
+                        + "Please, try again"));
+        if (driver.getPassword().equals(password)) {
+            return driver;
+        }
+        throw new AuthenticationException("Incorrect password. Please, try again");
```

Comment:
for the security purposes, I don't think we should inform user what exactly did they got wrong

Reply:
it depends on what is more important - help user or security)